linux_os/guide/system/accounts/accounts-restrictions/password_storage/accounts_password_last_change_is_in_past/rule.yml

```diff
@@ -0,0 +1,33 @@
+documentation_complete: true
+
+title: 'Ensure all users last password change date is in the past'
+
+description: |-
+    All users should have a password change date in the past.
+
+rationale: |-
+    If a user recorded password change date is in the future then they could
+    bypass any set password expiration.
+
+severity: medium
+
+identifiers:
+    cce@sle12: CCE-92330-0
+    cce@sle15: CCE-92504-0
+
+references:
+    cis@sle12: 5.4.1.6
+    cis@sle15: 5.4.1.6
+
+ocil_clause: 'any interactive user password that has last change time in the future'
+
+ocil: |-
+    Verify that the interactive user account passwords last change time is not in the future
+    The following command should return no output
+    <pre>$ sudo expiration=$(cat /etc/shadow|awk -F ':' '{print $3}');
+    for edate in ${expiration[@]}; do if [[ $edate > $(( $(date +%s)/86400 )) ]];
+    then echo "Expiry date in future";
+    fi; done </pre>
+
+fixtext: |-
+    Forcibly expire account is one possible solution to make sure that the policy will be followed
```

Comment:
Although a remediation for this rule would be possible to be automated by Bash and Ansible, I totally agree to not do that in order to avoid any system disruption. However, it would be great to include a [`warning`](https://complianceascode.readthedocs.io/en/latest/manual/developer/04_style_guide.html#rule-sections) informing the reason there is no automated remediation for that. This way user can see the warning in the guide.

Reply:
:+1:  added more descriptive fixtext and warning for automatic remediation absence in b59dece